### ISSUE TITLE
Wrong units for timestep

### ISSUE THREAD

**Issue by jchodera** [choderalab/perses#727]
I'm testing the latest `fixing-tests` branch from @hannahbrucemacdonald and noticed this odd output in the logs:
```
2020-08-26 20:51:34,653:(0.00s):root:   Pulling receptor (as pdb or mol2)...
2020-08-26 20:51:34,653:(0.00s):root:   detected ligand file: 2020-08-20-benzotriazoles-dockscores-x10876.sdf
2020-08-26 20:51:34,654:(0.00s):root:   old ligand index: 1; new ligand index: 0
2020-08-26 20:51:34,654:(0.00s):root:   setting up forcefield files...
2020-08-26 20:51:34,654:(0.00s):root:   timestep: 4.0 fs**2.
```
The timestep seems to have the wrong units!

It appears that [these lines](https://github.com/choderalab/perses/blob/ffec060401850df2b7c029af60415ae4b203b9ce/perses/app/setup_relative_calculation.py#L355-L357
) are taking a default timestep that already has fs units and multiply by fs:
```python
    if "timestep" in setup_options:
        timestep = setup_options['timestep'] * unit.femtoseconds
        _logger.info(f"\ttimestep: {timestep}.")
```
Note that OpenMM does not do any checking that dimensions are as expected---this is a shortcoming of the Python API. The effect is that units are stripped by multiplying whatever is needed to get into default OpenMM units, which are picoseconds. That means a `4.0 fs**2` timestep will be converted to `4e-6 ps`, presumably for equilibration, which is definitely _not_ what we want.

@dominicrufa @hannahbrucemacdonald Can you fix? 

We should also add checks to any case where you expect an option to have or lack units to make sure the units are as expected and the dimensionality matches your expectation.

**Comment by jchodera** [comment]
I think the issue is [here](https://github.com/choderalab/perses/blob/98fba4444aded9488e7e96cc6b9cbbf8e1809671/perses/app/fah_generator.py#L126):
```python
                      timestep=4.0 * unit.femtosecond,
```
but this is really a symptom of a problematic API that doesn't do any unit checking.

We can rearchitect the API later, but for now, we should aggressively check any assumptions within each code block to make sure that we don't pass the wrong things back and forth.

**Comment by hannahbrucemacdonald** [comment]
Closed by #729
